roles/openshift_certificate_expiry/test/test_fakeopensslclasses.py

```diff
@@ -1,82 +1,78 @@
+@pytest.fixture(scope='module')
+def fake_server_cert(server_cert_file):
+    server_cert = sh.openssl.x509('-in', str(server_cert_file), '-text')
+    return FakeOpenSSLCertificate(server_cert)
 
-    def test_FakeOpenSSLCertificate_get_serial_number(self):
-        """We can read the serial number from the cert"""
-        self.assertEqual(11, self.fake_cert.get_serial_number())
 
-    def test_FakeOpenSSLCertificate_get_notAfter(self):
-        """We can read the cert expiry date"""
-        expiry = self.fake_cert.get_notAfter()
-        self.assertEqual('20190207181935Z', expiry)
+@pytest.mark.parametrize('attribute', ['get_serial_number', 'get_notAfter'])
+def test_attributes(client_cert, fake_client_cert, server_cert, fake_server_cert, attribute):
+    assert getattr(client_cert, attribute)() == getattr(fake_client_cert, attribute)()
+    assert getattr(server_cert, attribute)() == getattr(fake_server_cert, attribute)()
 
-    def test_FakeOpenSSLCertificate_get_sans(self):
-        """We can read Subject Alt Names from a cert"""
-        ext = self.fake_san_cert.get_extension(0)
 
-        if ext.get_short_name() == 'subjectAltName':
-            sans = str(ext)
+def test_get_subject(client_cert, fake_client_cert, server_cert, fake_server_cert):
+    c_subjects = client_cert.get_subject().get_components()
+    c_subj = ', '.join(['{}:{}'.format(x, y) for x, y in c_subjects])
+    f_subjects = fake_client_cert.get_subject().get_components()
+    f_subj = ', '.join(['{}:{}'.format(x, y) for x, y in f_subjects])
+    assert c_subj == f_subj
 
-        self.assertEqual('DNS:kubernetes, DNS:kubernetes.default, DNS:kubernetes.default.svc, DNS:kubernetes.default.svc.cluster.local, DNS:m01.example.com, DNS:openshift, DNS:openshift.default, DNS:openshift.default.svc, DNS:openshift.default.svc.cluster.local, DNS:172.30.0.1, DNS:192.168.122.241, IP Address:172.30.0.1, IP Address:192.168.122.241', sans)
+    s_subjects = server_cert.get_subject().get_components()
+    s_subj = ', '.join(['{}:{}'.format(x, y) for x, y in s_subjects])
+    f_subjects = fake_server_cert.get_subject().get_components()
+    f_subj = ', '.join(['{}:{}'.format(x, y) for x, y in f_subjects])
+    assert s_subj == f_subj
 
-    def test_FakeOpenSSLCertificate_get_sans_no_sans(self):
-        """We can tell when there are no Subject Alt Names in a cert"""
-        with self.assertRaises(IndexError):
-            self.fake_cert.get_extension(0)
 
-    def test_FakeOpenSSLCertificate_get_subject(self):
-        """We can read the Subject from a cert"""
-        # Subject: O=system:nodes, CN=system:node:m01.example.com
-        subject = self.fake_cert.get_subject()
-        subjects = []
-        for name, value in subject.get_components():
-            subjects.append('{}={}'.format(name, value))
+def test_subject_alt_names(client_cert, fake_client_cert, server_cert, fake_server_cert):
+    client_san = None
+    for i in range(client_cert.get_extension_count()):
```

Comment:
I agree on preferring to iterate over extensions, but this is a limitation of pyopenssl, only `get_extension(index)` and `get_extension_count()` are available.